Feature `unstable-provider-api` + `alloc` + !`std` compile fails

# Steps to reproduce

```console
$ cargo new snafu-test
    Creating binary (application) `snafu-test` package
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

$ cd snafu-test

$ cargo add snafu@0.8.8 --no-default-features -F unstable-provider-api -F alloc -F rust_1_81
    Updating crates.io index
      Adding snafu v0.8.8 to dependencies
             Features:
             + alloc
             + rust_1_61
             + rust_1_65
             + rust_1_81
             + unstable-provider-api
             - backtrace
             - backtraces-impl-backtrace-crate
             - futures
             - futures-core-crate
             - futures-crate
             - guide
             - internal-dev-dependencies
             - pin-project
             - std
             - unstable-core-error
             - unstable-try-trait
    Updating crates.io index
     Locking 7 packages to latest Rust 1.89.0 compatible versions

$ cargo +nightly build -q
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `std`
   --> /home/nksm/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/snafu-0.8.8/src/report.rs:248:17
    |
248 |                 std::writeln!(f, "\nBacktrace:\n{}", bt)?;
    |                 ^^^ use of unresolved module or unlinked crate `std`
    |
note: found an item that was configured out
   --> /home/nksm/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/snafu-0.8.8/src/lib.rs:276:14
    |
275 | #[cfg(feature = "std")]
    |       --------------- the item is gated behind the `std` feature
276 | extern crate std;
    |              ^^^

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `std`
    --> /home/nksm/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/snafu-0.8.8/src/lib.rs:1642:38
     |
1642 | #[snafu(provide(opt, ref, chain, dyn std::error::Error => source.as_deref()))]
     |                                      ^^^ use of unresolved module or unlinked crate `std`
     |
     = help: if you wanted to use a crate named `std`, use `cargo add std` to add it to your `Cargo.toml`
help: consider importing this module
     |
 268 + use core::error;
     |
help: if you import `error`, refer to it directly
     |
1642 - #[snafu(provide(opt, ref, chain, dyn std::error::Error => source.as_deref()))]
1642 + #[snafu(provide(opt, ref, chain, dyn error::Error => source.as_deref()))]
     |

For more information about this error, try `rustc --explain E0433`.
error: could not compile `snafu` (lib) due to 2 previous errors

```

The first error has been reported as #513

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -136,6 +136,7 @@ rust_nightly_no_std_test_task:
     - rustc --version
   provider_api_test_script:
     - cargo +nightly build --target thumbv6m-none-eabi --no-default-features --features=rust_1_81,unstable-provider-api
+    - cargo +nightly build --target thumbv6m-none-eabi --no-default-features --features=rust_1_81,alloc,unstable-provider-api
   before_cache_script: rm -rf $CARGO_HOME/registry/index
 
 nightly_test_task:

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1639,7 +1639,7 @@ macro_rules! location {
 #[snafu(crate_root(crate))]
 #[snafu(whatever)]
 #[snafu(display("{message}"))]
-#[snafu(provide(opt, ref, chain, dyn std::error::Error => source.as_deref()))]
+#[snafu(provide(opt, ref, chain, dyn crate::Error => source.as_deref()))]
 #[cfg(any(feature = "alloc", test))]
 pub struct Whatever {
     #[snafu(source(from(Box<dyn crate::Error>, Some)))]

